### ISSUE TITLE
[Docusaurus] `{=mdx}` code block seems broken (Quarto 1.4 with Docusaurus v3)

### ISSUE THREAD

**Issue by eitsupi** [quarto-dev/quarto-cli#8333]
### Bug description

~~If we use mdx in mdx files, it doesn't seem necessary to write it as code blocks.~~

` ``` mdx-code-block` should be changed to ` ```mdx-code-block`

### Steps to reproduce

````qmd
---
title: foo
format: docusaurus-md
---

```{=mdx}
export const Highlight = ({children, color}) => (
  <span
    style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
    {children}
  </span>
);

<Highlight color="#25c2a0">Docusaurus GREEN</Highlight> and <Highlight color="#1877F2">Rams blue</Highlight> are my favorite colors.

I can write **Markdown** alongside my _JSX_!
```
````

### Expected behavior

````mdx
---
title: foo
format: docusaurus-md
---



```mdx-code-block
export const Highlight = ({children, color}) => (
  <span
    style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
    {children}
  </span>
);

<Highlight color="#25c2a0">Docusaurus GREEN</Highlight> and <Highlight color="#1877F2">Rams blue</Highlight> are my favorite colors.

I can write **Markdown** alongside my _JSX_!
```
````

### Actual behavior

````mdx
---
title: foo
format: docusaurus-md
---




``` mdx-code-block
export const Highlight = ({children, color}) => (
  <span
    style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
    {children}
  </span>
);

<Highlight color="#25c2a0">Docusaurus GREEN</Highlight> and <Highlight color="#1877F2">Rams blue</Highlight> are my favorite colors.

I can write **Markdown** alongside my _JSX_!
```
````

### Your environment

- Ubuntu 22.04

### Quarto check output

```sh
$ quarto check   
Quarto 1.4.545
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.545
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Tex:  (not detected)

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /usr/local/python/current/bin/python3
      Jupyter: 5.7.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.2
      Path: /usr/local/lib/R
      LibPaths:
        - /workspaces/querying-with-prql/renv/library/R-4.3/x86_64-pc-linux-gnu
        - /usr/local/lib/R/library
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
```

**Comment by cderv** [comment]
This `mdx-code-block` syntax was added in b761219fc0f45b7ab2b7301f20909b68a0a134ee 

I believe this happened before the change to `.mdx` extension in e86312cfd0f4a0002197e955ada32730482fc36a 

I don't have much context but looking at the docusaurus doc, it seems to be interesting syntax related to mdx support in some situation with specific tools, according to doc at https://docusaurus.io/docs/i18n/crowdin#mdx

Does the use of code block `mdx-code-block` creates an issue with Docusuarus rendering ? 

Doc says it should be 
> interpreted by Docusaurus as regular JSX (as if it was not wrapped by any code block)

@jjallaire do you remember about b761219fc0f45b7ab2b7301f20909b68a0a134ee and using 
````lua
function RawBlock(el)
  if el.format == 'mdx' then
    return pandoc.CodeBlock(el.text, pandoc.Attr("", {"mdx-code-block"}))

**Comment by eitsupi** [comment]
> Does the use of code block `mdx-code-block` creates an issue with Docusuarus rendering ?

In Docusaurus v3.0.1, which I tried, the `mdx-code-block` code block seems to be displayed on the website as a code block.

**Comment by eitsupi** [comment]
Thanks for the link to the documentation. This seems to be supported by Docusaurus 3.1.0, so maybe it's a bug of Docusaurus...

**Comment by cderv** [comment]
I understand the Docusaurus V3 is a big change, especially about MDX V1 to V3 : https://docusaurus.io/docs/migration/v3#upgrading-mdx

I am not sure exactly which version we support but I would say it is Docusaurus V2 because v3 is out since 31 October 2023 so not that long. So I would not surprise to find issues. @cscheid did we do specific work for Docusaurus V3 and upgrade to MDX v1 to v3 ? 

I am not sure and we probably need to check what Quarto is producing and adapt.

**Comment by eitsupi** [comment]
Now I know what the problem is.
Spaces before the infostring are not allowed.
In other words, it should be ` ```mdx-code-block`, not ` ``` mdx-code-block`

![image](https://github.com/quarto-dev/quarto-cli/assets/50911393/56cb85a1-f6da-411c-aecf-4e0eed3a9eb9)

v.s.

![image](https://github.com/quarto-dev/quarto-cli/assets/50911393/fd238bdd-6670-4d69-bb33-c44103bd1c2b)

**Comment by krishaamer** [comment]
> `` ``` mdx-code-block `` should changed be `` ```mdx-code-block ``

I can confirm on my project (Quarto 1.4.545 and Docusaurus 3.1.0 on MacOS Ventura) that removing the additional space after ``` immediately fixes the rendering. Hope this gets fixed soon. @cderv

**Comment by eitsupi** [comment]
Or, should it be modified to ignore the space before the infostring on the Docusaurus side (mdx side)?

**Comment by cderv** [comment]
Thanks for finding this. 

It seems they could indeed add support for space in their regex. I believe this is common and they support it for other  codeblocks. 

On our side, the space is added by Pandoc directly, as this is the Markdown syntax created from Lua 

````markdown
> pandoc -f native -t markdown_strict+fenced_code_blocks+backtick_code_blocks
CodeBlock ("",["mdx-code-block"],[]) "content"

``` mdx-code-block
content
```
````

It could also be seen as an improvement on Pandoc's side because they do not document with space in their doc (https://pandoc.org/MANUAL.html#fenced-code-blocks) but they do consider space as the created syntax if I try round-trip 
````
> pandoc -f markdown -t markdown
```haskell
qsort [] = []
```

``` haskell
qsort [] = []
```
````

From their doc: 

> If the fenced_code_attributes extension is disabled, but input contains class attribute(s) for the code block, the first class attribute will be printed after the opening fence as a bare word.

To fix this in Pandoc we would need to create ourself raw markdown instead of a CodeBlock at 
https://github.com/quarto-dev/quarto-cli/blob/9e35ab3f1bf38338c10e8b5b625bee60184165b8/src/resources/extensions/quarto/docusaurus/docusaurus.lua#L52-L56

**Comment by eitsupi** [comment]
Thanks.

IIUC, this is a bug that occurred between Docusaurus v2 and v3 and can convince them to ignore the space.

Until tomorrow I don't have time to do any more research or report or PR to the docusaurus repo, but I'll take a look at this on Saturday if someone doesn't.

**Comment by cderv** [comment]
I opened a PR with a change that should fix this - But this indeed maybe on docusaurus itself to have broader support.

**Comment by eitsupi** [comment]
FYI, the issue of Docusaurus has been fixed by facebook/docusaurus#9776
Maybe v3.1.1 includes that...
